ApollosProject/apollos-prototype
# Adds `AuthLoadingSwitch` to Navigate different directions, depending on if a user is logged in.

## DESCRIPTION

### What does this PR do, or why is it needed?


Adds and implements an AuthLoadingSwitchComponent, which navigates different ways if a user is logged in. When this component is the default route of a navigator, it will wait for a query to return describing if a user is logged in (including waiting for the cache to load) and then navigate to a named route described by the component. 

This is used to handle navigating correctly when the app mounts. To the "home" if the user is logged in, and to the "auth screen" if the user isn't logged in.

Here's an example of usage

```
const AuthLoading = (props) => <AuthLoadingSwitch loggedInRouteName="Home" loggedOutRouteName="Auth" LoadingIndicator={LoadingComponent}

const AppNavigator = createStackNavigator(
  {
    AuthLoading,
    Home,
    Auth,
  },
  {
    initialRouteName: 'AuthLoading',
  }
);
```

### What design trade-offs/decisions were made?

Chose to create this component in the `ui-auth` package to reduce the amount of code that needed to be copy/pasted.

### How do I test this PR?

1. Open the app when logged out, refresh the app, you should end up on the auth screen.
2. Open the app when logged in, refresh the app, you should end up on the tab view.

### What automated tests did you write?

Snapshot tests over new components.

## TODO:

- [x] References #519 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
